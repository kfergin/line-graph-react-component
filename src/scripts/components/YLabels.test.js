import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import YLabels from './YLabels';

describe('<YLabels/>', () => {
	const props = {
		labels: [],
		width: 200,
		height: 100
	};
	
	it('should render its empty snapshot', () => {
		const tree = renderer.create(
			<YLabels {...props}/>
		).toJSON();
		expect(tree).toMatchSnapshot();
	});

	// would you test for the number of labels?
	it('should render its snapshot with string labels', () => {
		const labels = ['first', 'second', 'third'];
		const tree = renderer.create(
			<YLabels {...{...props, labels}}/>
		).toJSON();
		expect(tree).toMatchSnapshot();
	});

	// would you test to make sure there isn't any text with empty labels?
	it('should render its snapshot with lines only (no text)', () => {
		const labels = Array.apply(null, {length: 3});
		const tree = renderer.create(
			<YLabels {...{...props, labels}}/>
		).toJSON();
		expect(tree).toMatchSnapshot();
	});

	// is this the right test for the componentWillReceiveProps part?
	it('should set a new timestamp in state when labels change', () => {
		const labels = ['first', 'second', 'third'];
		const component = mount(
			<YLabels {...{...props, labels}}/>
		);
		const instance = component.instance();
		const spySetState = jest.spyOn(instance, 'setState');
		component.setProps({labels});
		expect(spySetState).not.toHaveBeenCalled();
		component.setProps({labels: 1});
		expect(spySetState).toHaveBeenCalled();
	});
});