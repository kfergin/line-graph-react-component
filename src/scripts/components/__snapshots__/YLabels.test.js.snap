// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<YLabels/> should render its empty snapshot 1`] = `
<g
  className="y-labels"
>
  <g
    style={
      Object {
        "opacity": 1,
      }
    }
  />
</g>
`;

exports[`<YLabels/> should render its snapshot with lines only (no text) 1`] = `
<g
  className="y-labels"
>
  <g
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <g
      className="yLabel"
    >
      <line
        x1="0"
        x2={200}
        y1={100}
        y2={100}
      />
    </g>
    <g
      className="yLabel"
    >
      <line
        x1="0"
        x2={200}
        y1={50}
        y2={50}
      />
    </g>
    <g
      className="yLabel"
    >
      <line
        x1="0"
        x2={200}
        y1={0}
        y2={0}
      />
    </g>
  </g>
</g>
`;

exports[`<YLabels/> should render its snapshot with string labels 1`] = `
<g
  className="y-labels"
>
  <g
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <g
      className="yLabel"
    >
      <line
        x1="0"
        x2={200}
        y1={100}
        y2={100}
      />
      <text
        x="0"
        y={100}
      >
        first
      </text>
    </g>
    <g
      className="yLabel"
    >
      <line
        x1="0"
        x2={200}
        y1={50}
        y2={50}
      />
      <text
        x="0"
        y={50}
      >
        second
      </text>
    </g>
    <g
      className="yLabel"
    >
      <line
        x1="0"
        x2={200}
        y1={0}
        y2={0}
      />
      <text
        x="0"
        y={0}
      >
        third
      </text>
    </g>
  </g>
</g>
`;
